clients/java/zts/core/src/main/java/com/yahoo/athenz/zts/ZTSFileUtil.java

```diff
@@ -0,0 +1,105 @@
+package com.yahoo.athenz.zts;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.yahoo.athenz.auth.token.AccessToken;
+import com.yahoo.athenz.auth.token.jwts.JwtsSigningKeyResolver;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.*;
+
+public class ZTSFileUtil {
+
+    private static final Logger LOG = LoggerFactory.getLogger(ZTSFileUtil.class);
+
+    static private final String DEFAULT_ACCESS_TOKEN_DIR_PATH = "/var/lib/sia/tokens/";
+    static private final String ACCESS_TOKEN_PATH_PROPERTY = "zts.client.accesstoken.path";
+    static private final String ROLE_NAME_CONNECTOR = ",";
+    private JwtsSigningKeyResolver accessSignKeyResolver = null;
+    private ObjectMapper objectMapper = new ObjectMapper();
+    private Map<String, String> roleNameMap;
+
+    public ZTSFileUtil(JwtsSigningKeyResolver resolver) {
+        roleNameMap = new HashMap<>();
+        accessSignKeyResolver = resolver;
+    }
+
+    public void preload() {
+        String path = System.getProperty(ACCESS_TOKEN_PATH_PROPERTY, DEFAULT_ACCESS_TOKEN_DIR_PATH);
+        File dir = new File(path);
+
+        // preload the map from the domain, rolesname to file path
```

Comment:
we should add more comments here as far what is the expected directory structure otherwise it's not easy to follow why we're keep traversing, etc